tests/foreman/api/test_role.py

```diff
@@ -100,7 +101,47 @@ def test_positive_update(self):
 class CannedRoleTestCases(APITestCase):
     """Implements Canned Roles tests from API"""
 
-    @stubbed()
+    def create_org_admin_role(self, name=None, orgs=None, locs=None):
+        """Helper function to create org admin role for particular
+        organizations and locations by cloning 'Organization admin' role.
+
+        :param str name: The name of cloned Org Admin role
+        :param list orgs: The list of organizations for which the org admin is
+            being created
+        :param list locs: The list of locations for which the org admin is
+            being created
+        :return dict: This function returns dict representation of cloned role
+            data returned from 'clone' function
+        """
+        name = gen_string('alpha') if not name else name
+        default_org_admin = entities.Role().search(
+            query={'search': u'name="Organization admin"'})
+        org_admin = entities.Role(id=default_org_admin[0].id).clone(
+            data={
+                'role': {
+                    'name': name,
+                    'organization_ids': orgs or [],
+                    'location_ids': locs or []
+                }
+            }
+        )
+        self.assertEqual(org_admin['name'], name)
+        org_admin_role = entities.Role(id=org_admin['id']).read()
+        return org_admin_role
+
+    @classmethod
+    def setUpClass(cls):
+        """Creates two orgs and locations that will be used by all the
+        underlying tests
+        """
+        super(CannedRoleTestCases, cls).setUpClass()
+        # These two will be used as role taxonomies
+        cls.org1 = entities.Organization().create()
```

Comment:
I cannot divide them as filter org and role orgs, because in some test cases I used 'filter_orgs or org2' as role orgs as well. E.g to update the role orgs or so. So lets stick to this